### PR TITLE
UI: enable tracking of replaced transactions

### PR DESCRIPTION
**Pull request by kangarang**
addresses: #133 

previously, transactions which were "sped up" were orphaned in terms of the GUI's tx history (kept in local storage)

this pr enables tracking/replacing for such transactions.

routes/functions affected:
- `/send`: `send()`
- `/pool`: `addLiquidity()`, `removeLiquidity()`
- `/convert`: `convertTokens()`
- `/stake`: `stake()`, `withdraw()`
- `wrap()`
- `unwrap()`
- `approve()`

### CODE REVIEW

File: packages/frontend/src/hooks/useTransactionReplacement.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+              replacementReceipt: receipt,
+            }
+          }
+        }
+      }
+    },
+    [txHistory]
+  )
+
+  // useEffect(() => {
```
Comment (kangarang) [inline packages/frontend/src/hooks/useTransactionReplacement.ts:71]: this and other `watcher`-related code was an attempt to cleverly react to replaced txs. it was not a successful endeavor, and was commented out for documentation purposes. happy to remove if it's distracting.

File: packages/frontend/src/pages/Pools/PoolsContext.tsx [hunk 2]
```diff
@@ -596,7 +602,12 @@ const PoolsContextProvider: FC = ({ children }) => {
           })
         )
       }
-      await addLiquidityTx?.wait()
+
+      const res = await waitForTransaction(addLiquidityTx, { networkName: selectedNetwork.slug })
+      if (res && 'replacementTx' in res) {
+        setTxHash(res.replacementTx.hash)
```
[the remark this answers is not in the document]
Comment (miguelmota) [reply packages/frontend/src/pages/Pools/PoolsContext.tsx:608]: @kangarang that sounds good. The transaction model can be refactored at a later time. There's a lot of moving parts (eg ethers transaction, transaction model, txHistory, useTransactionReplacement, etc) which can be unified better in the future. I think the current PR solves the transaction replacement issue pretty well

**Comment by miguelmota** [comment]
@kangarang the source transaction was successfully detected as a replacement 🎉, 
but the destination transaction status failed to update due to an errorr.

 The error that it threw was `hop: no matching transaction 0x5e93149cd67a33da61d0f3bd1555fd2ccd7c45de3c7d2baa54f71d0e73e699eb`

DeepinScreenshot_select-area_20211029132251
DeepinScreenshot_select-area_20211029132311

**Comment by kangarang** [comment]
> @kangarang the source transaction was successfully detected as a replacement 🎉, but the destination transaction status failed to update due to an errorr.
> 
> The error that it threw was `hop: no matching transaction 0x5e93149cd67a33da61d0f3bd1555fd2ccd7c45de3c7d2baa54f71d0e73e699eb`
> 
> DeepinScreenshot_select-area_20211029132251 DeepinScreenshot_select-area_20211029132311

hmmm, what does your localStorage.recentTransactions look like?

also, i'm curious to know -- if you open the tx history list now, does the issue resolve itself?

**Comment by miguelmota** [comment]
@kangarang here's the additional info

DeepinScreenshot_select-area_20211029144013

DeepinScreenshot_select-area_20211029144033

DeepinScreenshot_select-area_20211029144042

Let me know if there's any other info needed for debugging

**Comment by miguelmota** [comment]
Let me know if this PR is ready to be tried again

**Comment by kangarang** [comment]
@miguelmota, ready for review!

**Review by miguelmota — APPROVED** [review]

**Comment by miguelmota** [comment]
works great @kangarang!

**Pull request merged**